### DISCUSSION TITLE
nccopy error with nczarr file

### DISCUSSION THREAD

**Discussion by CedricPenard** [Unidata/netcdf-c#2901] [Q&A]
Hi all !

I tried to convert a netcdf file into nczarr file with nccopy, but I experiment some error:

nccopy -d 4 MERGE_12H_13LEV_c_1_1_721_360_s_on.nc "file://tmp.zip#mode=nczarr,file,zip"
NetCDF: NCZarr error
Location: file /home/cedric/projets/Install/netcdf/netcdf-c/ncdump/nccopy.c; fcn copy line 2097

nccopy MERGE_12H_13LEV_c_1_1_721_360_s_on.nc "file://tmp.zip#mode=nczarr,file,zip"
NetCDF: Filter error: undefined filter encountered
Location: file /home/cedric/projets/Install/netcdf/netcdf-c/ncdump/nccopy.c; fcn copy_var_specials line 1147

Without zip:

nccopy -d 4 MERGE_12H_13LEV_c_1_1_721_360_s_on.nc "file://tmp.nczarr#mode=nczarr,file"
NetCDF: Filter error: undefined filter encountered
Location: file /home/cedric/projets/Install/netcdf/netcdf-c/ncdump/nccopy.c; fcn copy_var_specials line 1147

But this command works :
nccopy -d=4 MERGE_12H_13LEV_c_1_1_721_360_s_on.nc "file://tmp.nczarr#mode=nczarr,file"

But the nczarr file is not compressed (obviously).

What did I do wrong ?
Why it works whit -d=4 ?

nc-config --all

This netCDF 4.9.3-development has been built with the following features: 

  --cc            -> /usr/bin/cc
  --cflags        -> -I/home/cedric/projets/Install/install/include
  --libs          -> -L/home/cedric/projets/Install/install/lib -lnetcdf
  --static        -> -lHDF5::HDF5 -lhdf5::hdf5_hl -lm -lz -lzip -lblosc -lzstd -lbz2 -lsz -lcurl -lxml2

  --has-dap          -> yes
  --has-dap2         -> yes
  --has-dap4         -> yes
  --has-nc2          -> yes
  --has-nc4          -> yes
  --has-hdf5         -> yes
  --has-hdf4         -> no
  --has-logging      -> no
  --has-pnetcdf      -> no
  --has-szlib        -> yes
  --has-cdf5         -> yes
  --has-parallel4    -> no
  --has-parallel     -> no
  --has-nczarr       -> yes
  --has-zstd         -> yes
  --has-benchmarks   -> no
  --has-multifilters -> yes
  --has-stdfilters   -> deflate
  --has-quantize     -> no

  --prefix        -> /home/cedric/projets/Install/install
  --includedir    -> /home/cedric/projets/Install/install/include
  --libdir        -> /home/cedric/projets/Install/install/lib
  --plugindir     -> 
  --version       -> netCDF 4.9.3-development
  --build-system  -> cmake

There are 2 tests failed: 
99% tests passed, 2 tests failed out of 239

Total Test time (real) = 371.13 sec

The following tests FAILED:
	109 - nc_test4_tst_chunks (Failed)
	239 - nczarr_test_run_corrupt.sh (Failed)
Errors while running CTest
Output from these tests are in: /home/cedric/projets/Install/netcdf/netcdf-c/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
make: *** [Makefile:91: test] Error 8

Thanks

**Comment by DennisHeimbigner** [comment]
If you want to produce the output as a file tree, then you use this:
````"file://tmp.file#mode=nczarr,file,zip"````
If you want to produce a zip file then use this:
````"file://tmp.zip#mode=nczarr,file,zip"````
Note that if you unzipped the latter zip file, you should see
a file tree that looks like the first case.

**Reply by CedricPenard** [reply]
Ok this command works:
nccopy -d=4 MERGE_12H_13LEV_s_on.nc "file://tmp.zip#mode=nczarr,file,zip"

But tmp.zip has exactly the same size has tmp.nczarr
It is not compressed, only concatenated.

**Comment by DennisHeimbigner** [comment]
Sorry I was not clear.
When libnetcdf sees a modelist with two or more storage formats - file,zip in this case -- I only uses the first one.
So #mode=nczarr,file,zip is equivalent to #mode=nczarr,file. If you want to use the zip format, you
need to specify #mode=nczarr,zip

**Comment by CedricPenard** [comment]
Hi,
I tried:
nccopy -d=4 MERGE_12H_13LEV_s_on.nc "file://tmp.zip#mode=nczarr,zip"
I have a file tmp.zip
But it has the same size as nczarr directory.
Compression does not appear not to be applied.

**Comment by DennisHeimbigner** [comment]
Any chance you can send me your input file: MERGE_12H_13LEV_s_on.nc ?
